src/Map.js

```diff
@@ -141,19 +142,41 @@ export default class Map {
       <div id="greenstand-leaflet" style="position: relative;width: 100%;height: 100%;"></div>
       <div id="greenstand-map-spin" style="z-index: 999; position: absolute; width: 100%; top: 0px; left: 0px" ></div>
       <div id="greenstand-map-alert" style="z-index: 999; position: absolute; width: 100%; top: 0px; left: 0px" ></div>
-    `
-    domElement.appendChild(divContainer)
-    const mountTarget = document.getElementById('greenstand-leaflet')
-    const mountSpinTarget = document.getElementById('greenstand-map-spin')
-    const mountAlertTarget = document.getElementById('greenstand-map-alert')
-    this.spin = new Spin()
-    this.spin.mount(mountSpinTarget)
-    this.alert = new Alert()
-    this.alert.mount(mountAlertTarget)
-
-    this.map = this.L.map(mountTarget, mapOptions)
-    this.map.setView(this.initialCenter, this.minZoom)
-    this.map.attributionControl.setPrefix('')
+      <div id="greenstand-map-buttonPanel" style="z-index: 999; position: absolute; top: 0px; left:50%; transform: translateX(-50%)" ></div>
+    `;
+    domElement.appendChild(divContainer);
+    const mountTarget = document.getElementById('greenstand-leaflet');
+    const mountSpinTarget = document.getElementById('greenstand-map-spin');
+    const mountAlertTarget = document.getElementById('greenstand-map-alert');
+    const mountButtonPanelTarget = document.getElementById(
+      'greenstand-map-buttonPanel'
+    );
+    this.spin = new Spin();
+    this.spin.mount(mountSpinTarget);
+    this.alert = new Alert();
+    this.alert.mount(mountAlertTarget);
+    this.buttonPanel = new ButtonPanel();
+    this.buttonPanel.mount(mountButtonPanelTarget);
+
+    this.map = this.L.map(mountTarget, mapOptions);
+    this.map.setView(this.initialCenter, this.minZoom);
+    this.map.attributionControl.setPrefix('');
+
+    mountButtonPanelTarget.addEventListener('click', (e) => {
+      if (e.target.id === 'right-arrow') {
+        try {
+          this.goNextPoint();
+        } catch (e) {
+          log.warn('go next failed', e);
+        }
+      } else if (e.target.id === 'left-arrow') {
+        try {
+          this.goPrevPoint();
+        } catch (e) {
+          log.warn('go prev failed', e);
+        }
+      }
+    });
```

Comment:
Do you think it is good that we move these lines to the button panel and expose: onClickNext, onClickPrev,

Reply:
I'm considering add an option to `disablePrevNextButton`

Reply:
I'm not sure what you mean. The way I did it exposes these functions? How should I go about not exposing them? This repo is a bit advanced for me, just started working on it for fun and for the challenge.

Reply:
@VWRoli no no you are doing great, I just want that you move the code above into the ButtonPanel, so the behavior of it gets encapsulated into the ButtonPanel component, I think it makes more sense, for example, I open a new option called: 
new Map({disableButtonPanel})

then, in the current code, we need to take care of two places:
No1.
if(dsableButtonPanel){
    new ButtonPanel()...
}

No.2:
if(disableButtonPanel){
  buttonPanelTarget.addListner(....)
}

So it's getting cumbersome, in my mind, hiding the details (encapsulation) is the major benefit of using the component like ButtonPanel

Reply:
oh, maybe you need to expose events on the ButtonPanel:
new ButtonPanel({
onNext: this.goNextPoint,
onPrev: this.goPrevPoint,
})

Reply:
I understand now. I will see what I can do about it.